app/gui2/src/stores/suggestionDatabase/index.ts

```diff
@@ -63,21 +63,22 @@ class Synchronizer {
         }
       })
     })
+    lsRpc.on('executionContext/executionComplete', async () => {
+      const groups = await lsRpc.getComponentGroups()
+      this.groups.value = groups.componentGroups.map(
+        (group): Group => ({
+          name: group.name,
+          ...(group.color ? { color: group.color } : {}),
+          project: group.library as QualifiedName,
+        }),
+      )
+    })
```

Comment:
I feel checking for groups on every execution is too often. The groups affect how suggestions in suggestion db are affected: perhaps we could just refresh the list only when one second passed since last received suggestion db update.

Also, our suggestion db entries keep their group's indices instead of names for the sake of being faster, but changing groups can invalidate those indices. We can keep groups in map instead. Or just load groups on start and not care about updates (the simplest solution).

Reply:
@farmaazon ah, originally i had it only load groups on start, because I wasn't sure whether the list of groups ever needs to be refreshed